src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -557,9 +679,45 @@ def _get_env_and_group_from_log_analytics(
                 env_list = [e for e in env_list if e["location"] == location]
             if env_list:
                 # TODO check how many CA in env
-                env_details = parse_resource_id(env_list[0]["id"])
-                env.set_name(env_details["name"])
-                resource_group.name = env_details["resource_group"]
+                # set_name() with parse name and resource_group from environment id
+                env.set_name(env_list[0]["id"])
+                resource_group.name = env.resource_group.name
+
+
+def _get_custom_location_and_extension_id_and_location_from_cluster(
+        cmd,
+        env
+):
+    #  if connected cluster have one custom location (with ext namespace) bind to the container app ext, then use it.
+    if env.custom_location.name is None and env.custom_location.connected_cluster_id is None:
+        raise ValidationError(
+            "Please specify which connected-cluster-id or custom location you want to create the Connected environment.")
+
+    if env.custom_location.connected_cluster_id and env.custom_location.name is None:
+        connected_cluster = get_connected_k8s(cmd, env.custom_location.connected_cluster_id)
+        env.custom_location.location = connected_cluster.location
+        if env.custom_location.resource_group is None or env.custom_location.resource_group.name is None:
+            env.custom_location.resource_group = ResourceGroup(cmd, connected_cluster.resource_group, connected_cluster.location)
+
+        extension_list = list_cluster_extensions(cmd, connected_cluster_id=env.custom_location.connected_cluster_id)
+        extension = None
+        for e in extension_list:
+            if e.extension_type.lower() == "Microsoft.App.Environment".lower():
+                extension = e
+                break
+
+        if extension:
+            env.custom_location.cluster_extension_id = extension.id
+            custom_location_list = list_custom_location(cmd,
+                                                        connected_cluster_id=env.custom_location.connected_cluster_id)
+            if len(custom_location_list) == 0:
+                env.custom_location.namespace = extension.scope.cluster.release_namespace
+            for c in custom_location_list:
+                if extension.id in c.cluster_extension_ids and extension.scope.cluster.release_namespace == c.namespace:
```

Comment:
The extension is get from `list_cluster_extensions(cmd, connected_cluster_id=env.custom_location.connected_cluster_id)`, and the  custom location list is aslo get from `custom_location_list = list_custom_location(cmd,
                                                        connected_cluster_id=env.custom_location.connected_cluster_id)`, so the extension and custom location have to be in one connected cluster, right?

then `extension.id in c.cluster_extension_ids` is unused?

Reply:
We need to find custom locations with extension of 'Microsoft.App.Environment'.
Connected cluster Id have mutiple extensions.
CustomLocation.cluster_extension_ids may not contain extension with type 'Microsoft.App.Environment'.